fission/src/ui/components/ViewCube.tsx

```diff
@@ -0,0 +1,895 @@
+                context.textBaseline = "middle"
+                context.fillText(label, 128, 128)
+
+                const texture = new THREE.CanvasTexture(canvas)
+                const labelMaterial = new THREE.SpriteMaterial({ map: texture })
+                const labelSprite = new THREE.Sprite(labelMaterial)
+                labelSprite.scale.set(0.8, 0.8, 1)
+                labelSprite.position.copy(direction.clone().multiplyScalar(lineLength / 2 + 0.4))
+
+                group.add(line)
+                group.add(labelSprite)
+                group.position.copy(position)
+
+                return group
+            }
+
+            const xAxis = createAxisLine(
+                0xff0000,
+                new THREE.Vector3(-1, 0, 0),
+                new THREE.Vector3(-1.3, -0.98, -0.98),
+                "X"
+            )
+            axisGroup.add(xAxis)
+
+            const yAxis = createAxisLine(0x00ff00, new THREE.Vector3(0, 1, 0), new THREE.Vector3(0.98, 1.3, -0.98), "Y")
+            axisGroup.add(yAxis)
+
+            const zAxis = createAxisLine(0x0000ff, new THREE.Vector3(0, 0, 1), new THREE.Vector3(0.98, -0.98, 1.3), "Z")
+            axisGroup.add(zAxis)
+
+            return axisGroup
+        }
+
+        const axisIndicators = createAxisIndicators()
+        scene.add(axisIndicators)
+        axisRef.current = axisIndicators
+
+        const ambientLight = new THREE.AmbientLight(0xffffff, 0.8)
+        scene.add(ambientLight)
+
+        const directionalLight = new THREE.DirectionalLight(0xffffff, 0.3)
+        directionalLight.position.set(5, 5, 5)
+        scene.add(directionalLight)
+
+        const directionalLight2 = new THREE.DirectionalLight(0xffffff, 0.2)
+        directionalLight2.position.set(-2, -2, -2)
+        scene.add(directionalLight2)
+
+        const animate = () => {
+            if (rendererRef.current && sceneRef.current && cameraRef.current) {
+                const mainCamera = World.SceneRenderer.mainCamera
+                const controls = World.SceneRenderer.currentCameraControls
+
+                if (mainCamera && cubeRef.current && controls instanceof CustomOrbitControls) {
+                    const coords = controls.getCurrentCoordinates()
+
+                    const camEuler = new THREE.Euler(coords.phi + Math.asin(1 / Math.sqrt(3)), coords.theta, 0, "YXZ")
+                    const camQuat = new THREE.Quaternion().setFromEuler(camEuler).invert()
+
+                    const offsetQuat = new THREE.Quaternion().setFromEuler(new THREE.Euler(0, Math.PI / 4, 0))
+
+                    cubeRef.current.quaternion.copy(offsetQuat).multiply(camQuat)
+
+                    if (axisRef.current) {
+                        axisRef.current.quaternion.copy(cubeRef.current.quaternion)
+                    }
+                }
+
+                rendererRef.current.render(sceneRef.current, cameraRef.current)
+            }
+            requestAnimationFrame(animate)
+        }
+
+        animate()
```

Comment:
Given that the view cube never gets unmounted IDK if it will come up but the animation frame is never cancelled here

Reply:
It can be hidden from the setting panel. Does this run into any issue by your evaluation?